src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -509,43 +551,79 @@ private object ReadNestedResourceInline(ODataResourceWrapper resourceWrapper, IE
         }
 
         private void ApplyResourceSetInNestedProperty(IEdmProperty nestedProperty, object resource,
-            ODataResourceSetWrapper resourceSetWrapper, ODataDeserializerContext readContext)
+            ODataResourceSetWrapperBase resourceSetWrapper, ODataDeserializerContext readContext)
         {
             Contract.Assert(nestedProperty != null);
             Contract.Assert(resource != null);
             Contract.Assert(readContext != null);
 
+            object value = null;
+
             if (readContext.IsDeltaOfT)
```

Comment:
As above; you should no longer need to special case navigation property in patch -- you should be able to replace 560-574 with:
  object value = ReadNestedResourceSetInline(resourceSetWrapper, nestedProperty.Type, readContext);
And remove the new ReadDeltaNestedResourceSetInline()